### PR TITLE
fix(wasm): time.Sleep and Sign issues

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed time.Sleep stops main thread on WebAssembly runtime
- fixed Sign issue on WebAssembly runtime

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- gosdk: https://github.com/0chain/gosdk/pull/413
- zboxcli: https://github.com/0chain/zboxcli/pull/190
- js-client-sdk :   https://github.com/0chain/js-client-sdk/pull/103

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/time_sleep_wasm  |
| zboxcli  | fix/time_sleep_wasm  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: zboxcore/sdk/sdk.go [hunk 1]
```diff
@@ -1119,7 +1120,9 @@ func smartContractTxnValueFee(sn transaction.SmartContractTxnData,
 	txn.TransactionFee = fee
 	txn.TransactionType = transaction.TxnTypeSmartContract
 
-	if err = txn.ComputeHashAndSign(client.Sign); err != nil {
+	if err = txn.ComputeHashAndSign(func(hash string) (string, error) {
+		return sys.Sign(hash, client.GetClient().SignatureScheme, client.GetClientSysKeys())
```
Comment (peterlimg) [inline zboxcore/sdk/sdk.go]: I saw several places that use this sign callback function, we can move this function to client.Sign, and use client.Sign here just like what it did before to make this cleaner.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: zboxcore/sdk/sdk.go (hunk 1 above)
Comment (cnlangzi) [reply zboxcore/sdk/sdk.go]: reverted

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/time_sleep_wasm  |
| zboxcli  | fix/time_sleep_wasm  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/break_upload_on_error  |
| gosdk  | staging  |
| zboxcli  | fix/break_upload_on_error  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**